src/app/chat/chat-room/chat-room.component.ts

```diff
@@ -120,13 +126,27 @@ export class ChatRoomComponent extends RouterEnter {
       team_member_id: null,
       participants_only: false
     };
+    this.messagePageNumber = 0;
+    this.messagePagesize = 20;
+    this.loadingMesageSend = false;
+    this.isTyping = false;
+    this.typingMessage = '';
   }
 
   private _validateRouteParams() {
-    const teamId = Number(this.route.snapshot.paramMap.get('teamId'));
-    this.selectedChat.team_id = teamId;
-    if (Number(this.route.snapshot.paramMap.get('teamMemberId'))) {
+    if (this.teamId) {
+      this.selectedChat.team_id = this.teamId;
+    } else {
+      const teamId = Number(this.route.snapshot.paramMap.get('teamId'));
+      this.selectedChat.team_id = teamId;
+    }
+    if (this.teamMemberId) {
+      this.selectedChat.team_member_id = this.teamMemberId;
+    } else if (Number(this.route.snapshot.paramMap.get('teamMemberId'))) {
       this.selectedChat.team_member_id = Number(this.route.snapshot.paramMap.get('teamMemberId'));
+    } else if (this.participantsOnly) {
+      this.selectedChat.is_team = true;
+      this.selectedChat.participants_only = this.participantsOnly;
     } else {
       this.selectedChat.is_team = true;
       this.selectedChat.participants_only = JSON.parse(this.route.snapshot.paramMap.get('participantsOnly'));
```

Comment:
if we break into something like this.
`if (this.teamMemberId) {
      this.selectedChat.team_member_id = this.teamMemberId;
    } else {
      this.selectedChat.team_member_id = Number(this.route.snapshot.paramMap.get('teamMemberId'));
    }
    if (this.participantsOnly) {
      this.selectedChat.is_team = true;
      this.selectedChat.participants_only = this.participantsOnly;
    } else {
      this.selectedChat.is_team = true;
      this.selectedChat.participants_only = JSON.parse(this.route.snapshot.paramMap.get('participantsOnly'));
    }`
`this.selectedChat.is_team` will be always true. because last else statement call always.
to fix that if did something like this.
`if (this.teamMemberId) {
      this.selectedChat.team_member_id = this.teamMemberId;
      return;
    } else {
      this.selectedChat.team_member_id = Number(this.route.snapshot.paramMap.get('teamMemberId'));
      return;
    }
    if (this.participantsOnly) {
      this.selectedChat.is_team = true;
      this.selectedChat.participants_only = this.participantsOnly;
    } else {
      this.selectedChat.is_team = true;
      this.selectedChat.participants_only = JSON.parse(this.route.snapshot.paramMap.get('participantsOnly'));
    }`
it will not check `participantsOnly` because if team member id is not there it will go to else block and add undefiend to `this.selectedChat.team_member_id` and `return;`.

what I want to do is,
if we have team member id, get it from @Input parameter or rout parameter then not checking for participants Only.
if we didn't have team member id as any parameter, then only get participants Only from @Input parameter or rout parameter and make `this.selectedChat.is_team = true`.